modules/__tests__/TestSequences/Pop.js

```diff
@@ -0,0 +1,33 @@
+import expect from 'expect'
+import execSteps from './execSteps'
+
+export default (history, done) => {
+  const steps = [
+    (location) => {
+      expect(location).toMatch({
+        pathname: '/'
+      })
+
+      history.push('/home')
+    },
+    (location, action) => {
+      expect(action).toEqual('PUSH')
+      expect(location).toMatch({
+        pathname: '/home'
+      })
+
+      history.pop()
+    },
+    (location, action) => {
+      expect(action).toEqual('POP')
+      expect(location).toMatch({
+        pathname: '/'
+      })
+      expect(history.entries).toMatch([{
+        pathname: '/',
+      }])
```

Comment:
Yes, my app uses `history.entries` and can’t do what it needs with just
the current `history.location`. Let me explain:

Among the different navigation components available in React Native we have
a navigation a state` :

```js
// Build navigation state
const navigationState = {
  index: 0,
  routes: [
    { key: 'page_1' },
    { key: 'page_2' },
  ],
}
// Build the navigation component with NavigationExperimental API
const Navigation = (props) =>  (
  <NavigationCardStack
    {...props}
    navigationState={navigationState}
  />
)
```

or:

```js
// Build navigation state
const routes = [
  { key: 'page_1' },
  { key: 'page_2' },
]
const navigationState = {
  initialRoute: routes[0]
  initialRouteStack: routes,
}
// Build the navigation component with <Navigator />
const Navigation = (props) => (
  <Navigator
    {...props}
    {...navigationState}
  />
)
```

In order to work with `history`, `NavigationExperimental` and `Navigator` need
to depend on `history.entries` properties.

Indeed, `react-native` offers us two types of “back methods”. The first one allows
us to transition back and unmount the current scene. It means that we can't then
come back to this scene. The other one allow us to  jump backward without unmounting
the current scene and later jump forward to this scene. 

If we come back to my example,  in order to check these differences,
the current `history.location` is not enough, I need to analyze `history.entries`.

Here is a more detailed example:

First at all, we build the initial state.
```js
const initialHistory = {
  ...history,
  index: 1,
  entries: [{ pathanme: '/scene1' }, { pathname: '/scene2' }],
}
const initialNavigationState = {
  index: 1,
  routes: [{ key: '/scene1' }, { scene: '/scene2' }]
}
```

**Case 1.** After the call of `history.pop()`, the navigator transitions back
and unmounts the current scene. After that, we can't go back to `/scene2`
with `history.goForward()` because the entry was deleted in history and
the scene was unmounted in navigator.
```js
const historyAfterPop = {
  index: 0,
  routes: [{ key: '/scene1' }],
}
const navigationStateAfterPop = {
  index: 0,
  routes: [{ key: '/scene1' }],
}
```

**Case 2** After the call of `history.goBack()`, the navigator jump backward
without unmounting the current scene. In this case, we can jump forward
to `/scene1` later. `/scene2` is always mounted in `react-native`’s navigator.
```js
const historyAfterGoBack = {
  index: 0,
  routes: [{ key: '/scene1' }, { key: '/scene2' }],
}
const navigationStateGoBack = {
  index: 0,
  routes: [{ key: '/scene1' }, { key: '/scene2' }],
}
```

I hope you will find this understandable. It would be incredible to be able to use `history`
with `react-native`! :tada:.